include/swift/SIL/MemAccessUtils.h

```diff
@@ -46,7 +46,47 @@
 
 namespace swift {
 
-// stripAddressAccess() is declared in InstructionUtils.h.
+/// Get the base address of a formal access by stripping access markers and
+/// borrows.
+///
+/// If \p V is an address, then the returned value is also an address
+/// (pointer-to-address is not stripped).
+SILValue stripAccessMarkers(SILValue V);
+
+/// Attempt to return the address corresponding to a variable's formal access
+/// by stripping indexing and address projections.
+///
+/// \p V must be an address.
+///
+/// Returns an address. If the a formal access was successfully identified, and
+/// access markers have not yet been removed, then the returned address is
+/// produced by a begin_access marker.
+///
+/// To get the base address of the formal access behind the access marker,
+/// either call stripAccessMarkers() on the returned value, or call
+/// getAccessedAddress() on \p V.
+///
+/// To identify the underlying storage object of the access, use
+/// findAccessedStorage() on either \p V or the returned address.
+SILValue getAddressAccess(SILValue V);
+
+/// Convenience for stripAccessMarkers(getAddressAccess(V)).
+SILValue getAccessedAddress(SILValue V);
+
+/// Return true if \p accessedAddress points to a let-variable.
+///
+/// let-variables are only written during let-variable initialization, which is
+/// assumed to store directly to the same, unaliased accessedAddress.
+///
+/// The address of a let-variable must be a formal access root. A 'let' member
+/// of a struct is *not* a let-variable, because it's memory may be written when
+/// formally modifying the outer struct. A let-variable is either an entire
+/// local variable, global variable, or class property (this is the definition
+/// of an address root).
+///
+/// The caller should derive the accessed address using
+/// stripAccessMarkers(getAccessedAddress(ptr)).
```

Comment:
If you are going to assert this internal to this function in the implementation, please say that the API has such a precondition in the interface explicitly!

Reply:
I fixed up this comment